src/renderer/apiFetch.ts

```diff
@@ -0,0 +1,79 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { apiActions } from "readium-desktop/common/redux/actions";
+import { TApiMethod, TApiMethodName } from "readium-desktop/main/api/api.type";
+import { TMethodApi, TModuleApi } from "readium-desktop/main/di";
+import { diRendererGet } from "readium-desktop/renderer/di";
+import { ApiLastSuccess } from "readium-desktop/renderer/redux/states/api";
+import { ReturnPromiseType } from "readium-desktop/typings/pierre";
+import { Unsubscribe } from "redux";
+import * as uuid from "uuid";
+
+export async function apiFetch<T extends TApiMethodName>(path: T, ...requestData: Parameters<TApiMethod[T]>) {
```

Comment:
The term `fetch` is overloaded ( https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API ), and I think this leads to confusion when reading the code (i.e. usages of the `apiFetch()` function).

Given that both `apiFetch()` and `apiRefresh()` are really about "store" ("dispatch" / "subscribe") and "state", can we find more suitable names that represent what the functions actually do?

Reply:
I like this both names !

what is your proposal ?

Reply:
I do not have a concrete proposal, because my understanding of the scope/responsibility of the two "api" functions is only superficial. I am merely highlighting the fact that the term "fetch" is overloaded. I can see that this is really about "store" ("dispatch" / "subscribe") and "state" ... but I am not sure how to formulate / abbreviate the function contract into a concise meaningful name.

Reply:
It's more over ipc communication with the usage of redux state

Reply:
Behind the scenes, Electron's IPC asynchronous events are used, but this is an "implementation detail" inside the underlying plumbing / system guts (in other words, we should not use "ipc" in the publicly-exposed function names).

From the perspective of the function callers, a Redux action corresponding to a well-defined "API" is dispatched, and a promised result is awaited. In that light, would it make sense to replace `apiFetch()` with `apiDispatch`? (I think so, because any Redux programmer would understand the notion of "dispatching" actions)

`apiRefresh()` does not actually "refresh" anything (imperatively-speaking). Instead, it is really about triggering calls to `apiDispatch()` based on the last successful API action, by listening / subscribing to store changes. So would it make sense to replace `apiRefresh()` with `apiSubscribe()`? (I think so, because the function returns the opposite `unsubscribe` function)

Reply:
the apiFetch function doesn't works like this. apiFetch ask a request with a path and get back the data, whereas dispatch, update the state with actual data. It's not the same thing. I don't think apiDispatch is a great name, at all. But in fact apiRefresh may be called apiSubscribe. The name refresh allow to perfect understand what this function execute the callback only when any api request match with api name array in first parameter. It's my opinion.

Reply:
> the apiFetch function doesn't works like this
Once again, I am just quoting the code, verbatim:
https://github.com/readium/readium-desktop/blob/6cbd90399ae4c9bcefed0880c8ced776bf57fa5b/src/renderer/apiFetch.ts#L27-L34
Anyway, I am happy to follow your guidance, this is why I asked you in the first place: to get your expert opinion.
Let's close this matter.

Reply:
Just for tracking purposes:
`apiRefresh()` becomes `apiSubscribe()` (because `Unsubscribe` is returned)